### ISSUE TITLE
Fail to pull image with 504 Gateway time-out and can not report the ExApp status

### ISSUE THREAD

**Issue by aef5748** [cloud-py-api/app_api#325]
### Describe the bug

I got two problems as below when I test deploy:
1. Can not pull image with 504 Gatewa time-out
```
Failed to pull image, GuzzleException occur: Server error: `POST http://nextcloud-appapi-dsp:2375/v1.41/images/create?fromImage=ghcr.io%2Fcloud-py-api%2Ftest-deploy%3Arelease` resulted in a `504 Gateway Time-out` response
```
![Screenshot from 2024-07-12 16-04-02](https://github.com/user-attachments/assets/06735ada-ae38-4196-873b-eae31c1e48c6)

2. Can't report the ExApp status to the Nextcloud instance.
```
TRACE:    172.18.0.4:54590 - HTTP connection lost
Try default url to report the init status: http://192.168.2.15:3000
[408] <request: PUT /ocs/v1.php/apps/app_api/apps/status/test-deploy>

ERROR occurred! Can't report the ExApp status to the Nextcloud instance.
TRACE:    172.18.0.4:54590 - ASGI [3] Completed
```
for detail in log file:
[nc_app_test-deploy_logs.txt](https://github.com/user-attachments/files/16190829/nc_app_test-deploy_logs.txt)

---
What can I do to solve them?

### Steps/Code to Reproduce

1. Install AppAPI
2. Setup docker socket proxy
3. Setup deploy daemons
4. Press Test deploy

### Expected Results

No error after test deploy

### Actual Results

1. Can not pull image with 504 gateway timeout
2. Can't report the ExApp status to the Nextcloud instance.

### Setup configuration

docker-compose
```
version: '3.7'
networks:
  nextcloud:
    name: nextcloud_proxy
    driver: bridge
    
volumes:
  nexecloud:
  db:
  appapi:

services:
  db:
    image: postgres:12-alpine
    container_name: Nextcloud-DB
    restart: always
    volumes:
      - /home/sw/Nextcloud/db:/var/lib/postgresql/data
    environment:
      - POSTGRES_DB=nextcloud
      - POSTGRES_USER=nextcloud
      - POSTGRES_PASSWORD=nextcloud
      - TZ=Asia/Taipei
    networks:
      - nextcloud

  app:
    image: nextcloud:29.0.3-apache
    container_name: Nextcloud
    restart: always
    ports:
      - 3000:80
    depends_on:
      - db
      - appapi
    links:
      - db
      - appapi
    volumes:
      - /home/sw/Nextcloud/web:/var/www/html
      - /etc/localtime:/etc/localtime:ro
    environment:
      - POSTGRES_DB=nextcloud
      - POSTGRES_HOST=db
      - POSTGRES_USER=nextcloud
      - POSTGRES_PASSWORD=nextcloud
    networks:
      - nextcloud
      
  appapi:
    image: ghcr.io/cloud-py-api/nextcloud-appapi-dsp:release
    container_name: nextcloud-appapi-dsp
    hostname: nextcloud-appapi-dsp
    restart: unless-stopped
    privileged: true
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
    environment:
      - NC_HAPROXY_PASSWORD=nextcloud
      - TIMEOUT_SERVER=1800s
      - TIMEOUT_CONNECT=180s
      - TIMEOUT_CLIENT=180s
    ports:
      - 127.0.0.1:2375:2375
    networks:
      - nextcloud
```

AppApi config
```
ExApps installed: 1
Name: docker_socket_proxy
Protocol: http
Host: nextcloud-appapi-dsp:2375
 
Deploy config
Docker network: nextcloud_proxy
Nextcloud URL: http://192.168.2.15:3000
HaProxy password: nextcloud
GPUs support: false
Compute device: CPU
```

**Comment by andrey18106** [comment]
@aef5748 HI! Did I understood correctly, that there are two errors occurred, first time failed to pull image, but later it was successful, but failed to reach out to Nextcloud?

**Comment by aef5748** [comment]
Hi @andrey18106 
Yes, here 2 errors occurred. 
When the appapi could not pull the image, I tried to pull the image from the cloud via docker pull, but still shows can not pull the image. So I clone the cloud-py-api/test-deploy and then build it myself.

I retested and deployed it from appapi, but it still shows that I can not pull the image. Then I retried removing ports on the nextcloud-appapi-dsp container but got the same error. 
I don't know what happened after several retries it can be passed. I saw it pull the cloud-py-api/test-deploy-cpu image and deploy it not cloud-py-api/test-deploy.

But the initialization phase got another error message "Can't report the ExApp status to the Nextcloud instance.".
The nextcloud container can be passed by ping from nextcloud-appapi-dsp. so the network is workable.
